lib/util/historical-buffer-manager.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import interfaces = require('../interface');
+
+// Internal class that holds the buffer array, a counter for overflow items and maxBufferLength
+class Buffer<T> {
+    data: T[];
+    overflow: number;
+    maxBufferLength: number;
+
+    get getObject(): interfaces.IBufferedData<T> {
+        return { buffer: this.data.slice(),
```

Comment:
As a style guide issue, I would propose that you use the style @ericvw proposes here in general (i.e. braces and properties don't share lines), with an exception for when you have just one property and you inline it all.